internal/k8s/controllers/tcp_route_controller_test.go

```diff
@@ -0,0 +1,170 @@
+	}
+)
+
+func TestTCPRouteSetup(t *testing.T) {
+	require.Error(t, (&TCPRouteReconciler{}).SetupWithManager(nil))
+}
+
+func TestTCPRoute(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		name          string
+		err           error
+		result        reconcile.Result
+		expectationCB func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager)
+	}{{
+		name: "get-error",
+		err:  errExpected,
+		expectationCB: func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager) {
+			client.EXPECT().GetTCPRoute(gomock.Any(), tcpRouteName).Return(nil, errExpected)
+		},
+	}, {
+		name: "deleted",
+		expectationCB: func(client *mocks.MockClient, reconciler *reconcilerMocks.MockReconcileManager) {
+			client.EXPECT().GetTCPRoute(gomock.Any(), tcpRouteName).Return(nil, nil)
+			reconciler.EXPECT().DeleteTCPRoute(gomock.Any(), tcpRouteName)
+		},
+	}} {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			client := mocks.NewMockClient(ctrl)
+			reconciler := reconcilerMocks.NewMockReconcileManager(ctrl)
+			if test.expectationCB != nil {
+				test.expectationCB(client, reconciler)
+			}
+
+			controller := &TCPRouteReconciler{
+				Context:        context.Background(),
+				Client:         client,
+				Log:            hclog.NewNullLogger(),
+				ControllerName: mockControllerName,
+				Manager:        reconciler,
+			}
+			result, err := controller.Reconcile(context.Background(), reconcile.Request{
+				NamespacedName: tcpRouteName,
+			})
+			if test.err != nil {
+				require.Error(t, err)
+				require.ErrorIs(t, err, test.err)
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, test.result, result)
+		})
+	}
+}
+
+func TestTCPRouteReferencePolicyToRouteRequests(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	serviceNamespace := gw.Namespace("namespace3")
+
+	backendObjRef := gw.BackendObjectReference{
+		Name:      gw.ObjectName("service"),
+		Namespace: &serviceNamespace,
+	}
+
+	httpRouteSpec := gw.HTTPRouteSpec{
+		Rules: []gw.HTTPRouteRule{{
+			BackendRefs: []gw.HTTPBackendRef{{
+				BackendRef: gw.BackendRef{
+					BackendObjectReference: backendObjRef,
+				},
+			}},
+		}},
+	}
+
+	tcpRouteSpec := gw.TCPRouteSpec{
+		Rules: []gw.TCPRouteRule{{
+			BackendRefs: []gw.BackendRef{{
+				BackendObjectReference: backendObjRef,
+			}},
+		}},
+	}
+
+	refPolicy := gw.ReferencePolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "ReferencePolicy"},
+		ObjectMeta: metav1.ObjectMeta{Namespace: "namespace3"},
+		Spec: gw.ReferencePolicySpec{
+			From: []gw.ReferencePolicyFrom{{
+				Group:     "gateway.networking.k8s.io",
+				Kind:      "TCPRoute",
+				Namespace: "namespace1",
+			}},
+			To: []gw.ReferencePolicyTo{{
+				Kind: "Service",
+			}},
+		},
+	}
+
+	client := gatewayclient.NewTestClient(
+		nil,
+		&gw.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tcproute",
```

Comment:
```suggestion
				Name:      "httproute",
```

Reply:
Thanks for catching this, assumed I had at least one of these mixups from all the renaming hah!